content/blog/intro-to-gamedev/index.md

```diff
@@ -0,0 +1,560 @@
+
+I thought we could take this chance to learn a little bit about web-based game development and create a small clone of the game.
+
+You can take a play the [finished game here](https://ignacioamigo.github.io/dinogame/bin/index.html) and [review the code here](https://github.com/IgnacioAmigo/dinogame). Keep in mind that the code is much more commented on this article since it's meant to be as explicit and educating as possible.
+
+Let's dive in!
+
+## Scope
+
+It's very easy to overshoot when scoping any piece of software, especially when you can clearly see many ways in which your software can be expanded which is the case for pretty much any kind of game (power-ups, sound effects, types of entities, etc.). In these kinds of contexts, (over-)engineering the code to be prepared for future implementations seems like the correct way to go.
+
+In this case, however, we have a pretty clear idea of what we want our game to look like: It's a simple infinite runner, where a couple of obstacles spawn from time to time and the score is directly based on how much time we can succesfully avoid them.
+
+It will be hard, but let's try to keep the code and program simple and pragmatic. There's value in making the code work more systematically but we are looking into low-friction implementation since we want this to work as an introduction to a couple of ideas and technologies. Furthermore, since we are learning some of these technologies, I'll try to be as explicit as possible while exploring some of their defining features (in general though, I would recommend diving deeper to each of the concepts).
+
+## Tech requirements
+
+Keeping the scope in mind, let's try to deduce and compress what we need from a technology perspective:
+
+- We want the game to run on the browser, which pretty much inevitably means we are going to use either Javascript, or something that ends up turning into Javascript.
+- We have to do game-specific tasks such as render images and process input. Luckily the browser gives us good support for both of them through things like WebGL and Javascript's input system respectively.
+- We would also like it to be as responsive as possible since it makes for a better user experience.
+
+Note that we don't really need brazing fast graphic drawing for this since the game will not render thousands of objects at once, so using WebGL is probably overkill. However, it's a good way to get into the topic and so we'll take the chance to do it that way. WebGL is fairly complicated to work with from scratch, so after doing a bit of research I have decided we can use PixiJS to render stuff a pretty simple set of Javascript libraries for a 2D renderer that runs over WebGL (which in turn means we are using hardware acceleration to render stuff). We can also go a step further and try to use it with Typescript, to get that sweet, sweet type-checking.
```

Comment:
PixiJS to render stuff a pretty simple set of Javascript  ?

Also, I recommend in the future you try to break these long lines into multiple lines, it makes it much easier for me to comment on a specific line. Adding a single break wont change the way this paragraph render and looks.

Reply:
jajajaja sweet sweet type checking you junky

Reply:
I figure this is a long enough post to justify adding a table of contents, what do you think?